### PR TITLE
[ADAM-2145] Add Spark 2.4.3 and Scala 2.12 to Jenkins build

### LINKED ISSUE #2145: Add Spark 2.4.3 and Scala 2.12 to Jenkins build

### PR DESCRIPTION
**Pull request by heuermh**
Fixes #2145

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -588,6 +588,11 @@
         <artifactId>adam-codegen-spark2_${scala.version.prefix}</artifactId>
         <version>${project.version}</version>
       </dependency>
+      <dependency>
+        <groupId>com.thoughtworks.paranamer</groupId>
```
Comment (akmorrow13) [inline pom.xml]: What are you using this for?

Comment (heuermh) [reply pom.xml]: More classpath woes
https://issues.apache.org/jira/browse/SPARK-26146
https://issues.apache.org/jira/browse/SPARK-26583

Comment (akmorrow13) [reply pom.xml]: Ahh I see

Comment (heuermh) [reply pom.xml]: It's been very frustrating dealing with the Spark folks, they keep going fast and breaking things without cleaning up any of the dependency mess left in the wake

**Comment by heuermh** [comment]
Jenkins, retest this please

**Comment by heuermh** [comment]
@fnothaft I'm very confused by the most recent Jenkins test failure.

It says that avro `GenericRecord` is missing from the classpath

```
Exception in thread "main" java.lang.NoClassDefFoundError: org/apache/avro/generic/GenericRecord
	at org.apache.spark.serializer.KryoSerializer.newKryo(KryoSerializer.scala:123)
	at org.apache.spark.serializer.KryoSerializerInstance.borrowKryo(KryoSerializer.scala:324)
	at org.apache.spark.serializer.KryoSerializerInstance.<init>(KryoSerializer.scala:309)
	at org.apache.spark.serializer.KryoSerializer.newInstance(KryoSerializer.scala:218)
	at org.apache.spark.broadcast.TorrentBroadcast$.blockifyObject(TorrentBroadcast.scala:288)
	at org.apache.spark.broadcast.TorrentBroadcast.writeBlocks(TorrentBroadcast.scala:127)
	at org.apache.spark.broadcast.TorrentBroadcast.<init>(TorrentBroadcast.scala:88)
	at org.apache.spark.broadcast.TorrentBroadcastFactory.newBroadcast(TorrentBroadcastFactory.scala:34)
	at org.apache.spark.broadcast.BroadcastManager.newBroadcast(BroadcastManager.scala:62)
	at org.apache.spark.SparkContext.broadcast(SparkContext.scala:1489)
	at org.apache.spark.rdd.NewHadoopRDD.<init>(NewHadoopRDD.scala:79)
	at org.apache.spark.SparkContext.$anonfun$newAPIHadoopFile$2(SparkContext.scala:1160)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)
	at org.apache.spark.SparkContext.withScope(SparkContext.scala:699)
	at org.apache.spark.SparkContext.newAPIHadoopFile(SparkContext.scala:1146)
	at org.bdgenomics.adam.rdd.ADAMContext.$anonfun$loadBam$1(ADAMContext.scala:1583)
...
Caused by: java.lang.ClassNotFoundException: org.apache.avro.generic.GenericRecord
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 43 more
```

but that class is present in the ADAM assembly jar

```
$ jar tvf adam-assembly/target/adam-assembly-spark2_2.12-0.27.0-SNAPSHOT.jar \
  | grep GenericRecord

   287 Tue May 21 14:15:12 CDT 2019 org/apache/avro/generic/GenericRecord.class
  5860 Tue May 21 14:15:12 CDT 2019 org/apache/avro/generic/GenericRecordBuilder.class
```

as `avro-1.8.2.jar` is pulled in in compile scope via bdg-formats 0.12.0

```
$ mvn dependency:tree
...
[INFO] --- maven-dependency-plugin:2.8:tree (default-cli) @ adam-assembly-spark2_2.12 ---
[INFO] org.bdgenomics.adam:adam-assembly-spark2_2.12:jar:0.27.0-SNAPSHOT
[INFO] \- org.bdgenomics.adam:adam-cli-spark2_2.12:jar:0.27.0-SNAPSHOT:compile
...
[INFO]    +- org.bdgenomics.bdg-formats:bdg-formats:jar:0.12.0:compile
[INFO]    |  \- org.apache.avro:avro:jar:1.8.2:compile
```

**Pull request merged**